src/main.ts

```diff
@@ -0,0 +1,137 @@
+                break;
+            case EventType.ISSUE_COMMENT:
+                await schemaOnIssueComment.validate(event_payload_text);
+                parsedSchema = schemaOnIssueComment.cast(event_payload_text);
+                account = parsedSchema.sender.login;
+                message = `**A Github issue comment was posted** by ${account} with content ${parsedSchema.comment.body}`;
+                url = parsedSchema.issue.html_url;
+                details = `Issue state: ${parsedSchema.issue.state}  - assignee: ${parsedSchema.issue.assignee}`;
+                break;
+            default:
+                if (event_name) {
+                    message = event_name.replace(/_/g, ".");
+                } else {
+                    message = "A GitHub Actions event has occurred"
+                }
+                url = `https://github.com/${repository}/actions`;
+        }
+
+        if (!details) {
+            details = core.getInput('details');
+        }
+    } catch (error) {
+        core.setFailed(error.message);
+    }
+    return new NotificationMessage(message, url, details);
+}
+
+
+async function notifyTeams(notificationMessage: NotificationMessage) {
+    const matches = msteams_webhook_url.match(/^https?\:\/\/([^\/?#]+)(.*)/i);
+    const hostname_match = matches && matches[1];
+    const path_match = matches && matches[2];
+
+    let req_data = JSON.stringify(require(jsonPath));
+    req_data  = req_data.replace(/GITHUB_WORKFLOW/g, `${workflow}`)
+                        .replace(/GITHUB_REPOSITORY/g, `${repository}`)
+                        .replace(/GITHUB_REF/g, `${branch}`)
+                        .replace(/GITHUB_TRIGGER_EVENT_DETAILS/g, `${notificationMessage.getDetails()}`)
+                        .replace(/GITHUB_TRIGGER_EVENT/g, `${notificationMessage.getMessage()}`)
+                        .replace(/GITHUB_EVENT_URL/g, `${notificationMessage.getUrl()}`)
+                        .replace(/GITHUB_STATUS/g, `${job_status}`);
+
+    const options = {
+        hostname: `${hostname_match}`,
+        port: 443,
+        path: `${path_match}`,
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': req_data.length
+        }
+    };
+
+    const req = https.request(options, (res) => {
+        if (res.statusCode >= 400) {
+            core.setFailed(`Action failed with status code  ${res.statusCode}`);
+        } else {
+            console.log(`statusCode: ${res.statusCode}`);
+        }
+
+        res.on('data', (d) => {
+            process.stdout.write(d)
+        })
+    });
+
+    req.on('error', (error) => {
+        console.error(error);
+        core.setFailed(`Action failed with error ${error}`);
+    });
+
+    req.write(req_data);
+    req.end();
+}
+
+parsedNotificationMessage(JSON.stringify(require(github_event_payload))).then((parsedMessage: NotificationMessage) => {
```

Comment:
`notifyTeams` already has the signature that you are using for you anonymous function here so it is redundant. It could be simplified to: 

```
parsedNotificationMessage(JSON.stringify(require(github_event_payload))).then(notifyTeams);
```
Having said that, I am not convinced that this composition is sound in its current form because I do not believe that the return type of `parseNotificationMessage` is _really_ `Promise<NotificationMessage>` - that seems unduly optimistic.